website/pages/docs/secrets/databases/custom.mdx

```diff
@@ -12,81 +12,166 @@ description: |-
-and is not required knowledge for day-to-day usage. If you don't plan on writing
-any plugins, we recommend not reading this section of the documentation.
-
 Please read the [Plugins internals](/docs/internals/plugins) docs for more
 information about the plugin system before getting started building your
 Database plugin.
 
 ## Plugin Interface
 
-All plugins for the database secrets engine must implement the same simple interface.
+All plugins for the database secrets engine must implement the same interface. This interface
+is found in `sdk/database/dbplugin/v5/database.go`
 
 ```go
 type Database interface {
+	// Initialize the database plugin. This is the equivalent of a constructor for the
+	// database object itself.
+	Initialize(ctx context.Context, req InitializeRequest) (InitializeResponse, error)
+
+	// NewUser creates a new user within the database. This user is temporary in that it
+	// will exist until the TTL expires.
+	NewUser(ctx context.Context, req NewUserRequest) (NewUserResponse, error)
+
+	// UpdateUser updates an existing user within the database.
+	UpdateUser(ctx context.Context, req UpdateUserRequest) (UpdateUserResponse, error)
+
+	// DeleteUser from the database. This should not error if the user didn't
+	// exist prior to this call.
+	DeleteUser(ctx context.Context, req DeleteUserRequest) (DeleteUserResponse, error)
+
+	// Type returns the Name for the particular database backend implementation.
+	// This type name is usually set as a constant within the database backend
+	// implementation, e.g. "mysql" for the MySQL database backend. This is used
+	// for things like metrics and logging. No behavior is switched on this.
 	Type() (string, error)
-	CreateUser(ctx context.Context, statements Statements, usernameConfig UsernameConfig, expiration time.Time) (username string, password string, err error)
-	RenewUser(ctx context.Context, statements Statements, username string, expiration time.Time) error
-	RevokeUser(ctx context.Context, statements Statements, username string) error
-	RotateRootCredentials(ctx context.Context, statements []string) (config map[string]interface{}, err error)
-	Init(ctx context.Context, config map[string]interface{}, verifyConnection bool) (saveConfig map[string]interface{}, err error)
+
+	// Close attempts to close the underlying database connection that was
+	// established by the backend.
 	Close() error
 }
 ```
 
-You'll notice the first parameter to a number of those functions is a
-`Statements` struct. This struct is used to pass the Role's configured
-statements to the plugin on function call. The struct is defined as:
+Each of the request and response objects can also be found in `sdk/database/dbplugin/v5/database.go`.
 
-```go
-type Statements struct {
-	Creation   []string
-	Revocation []string
-	Rollback   []string
-	Renewal    []string
-}
-```
+In each of the requests, you will see a `Statements` object. This object represents the set of
```

Comment:
```suggestion
In each of the requests, you will see at least 1 `Statements` object. This object represents the set of
```

nit: `UpdateUserRequest` has multiple, and the fact they weren't in the top level was slightly confusing for me initially.

Reply:
Good call. I also added a note here that `UpdateUserRequest` has them in sub-fields.